src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -112,6 +123,56 @@ class RegularSync(
       }
   }
 
+  /**
+    * Handles NewHashesMessege, should only cover this message when we are top of the chain
+    */
+  def handleNewBlockHashesMessages: Receive = {
+    case MessageFromPeer(NewBlockHashes(hashes), peerId) =>
+      val maybePeer = peersToDownloadFrom.find(peer => peer._1.id == peerId)
+      //we allow asking for new hashes when we are not syncing and we can download from specified peer and we are
+      //top of the chain
+      if (notDownloading() && topOfTheChain && maybePeer.isDefined) {
+        val (peer, _) = maybePeer.get
+        val hashesToCheck = hashes.take(syncConfig.maxNewHashes)
+
+        if (!containsAncientBlockHash(hashesToCheck)) {
+          val filteredHashes = getValidHashes(hashesToCheck)
+
+          if (filteredHashes.nonEmpty) {
+            val request = GetBlockHeaders(Right(filteredHashes.head.hash), filteredHashes.length, BigInt(0), reverse = false)
+            requestBlockHeaders(peer, request)
+            resolvingBranches = false
```

Comment:
Should we maybe also cancel the scheduled `ResumeRegularSync` if there is one? If I understood correctly, when processing the requested headers we should always end up scheduling a `ResumeRegularSync`.
Also, as in `scheduleResume()` we don't cancel the previously scheduled `ResumeRegularSync` we could end up with 2 of them scheduled (as would happen if the [request for the headers associated with the hashes fail](https://github.com/input-output-hk/mantis/blob/feature/new_block_hashes/src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala#L193)) and I'm not sure exactly if there could potentially be any issue with how we handle that.

Reply:
I think you have valid point here and we should cancel it. 👍